src/Microsoft.Management.Deployment/Helpers.cpp

```diff
@@ -78,24 +84,34 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             // Get the caller process id and use it to check if the caller has permissions to access the feature.
             winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus status = winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus::DeniedBySystem;
 
-            auto capability = winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapability::CreateWithProcessIdForUser(nullptr, GetStringForCapability(requiredCapability), callerProcessId);
-            status = capability.CheckAccess();
-
-            allowed = (status == winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus::Allowed);
-        }
-        else
-        {
-            // If AppCapability is not present, require at least medium IL callers
-            auto requiredIntegrityLevel = AppInstaller::Security::IntegrityLevel::Medium;
+            winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapability capability{ nullptr };
 
-            if (callerProcessId != GetCurrentProcessId())
+            try
             {
-                allowed = AppInstaller::Security::IsCOMCallerIntegrityLevelAtLeast(requiredIntegrityLevel);
+                capability = winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapability::CreateWithProcessIdForUser(nullptr, GetStringForCapability(requiredCapability), callerProcessId);
             }
-            else
+            catch (const winrt::hresult_invalid_argument&)
             {
-                allowed = AppInstaller::Security::IsCurrentIntegrityLevelAtLeast(requiredIntegrityLevel);
             }
+
+            if (capability)
+            {
+                status = capability.CheckAccess();
+
+                return (status == winrt::Windows::Security::Authorization::AppCapabilityAccess::AppCapabilityAccessStatus::Allowed);

```

Comment:
The method expects an hresult return type

Reply:
Ugh, good catch, fixed.